pygen/pygen_src/riscv_directed_instr_lib.py

```diff
@@ -0,0 +1,103 @@
+"""
+Copyright 2020 Google LLC
+Copyright 2020 PerfectVIPs Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+"""
+
+import vsc
+from enum import Enum, auto
+from pygen_src.riscv_instr_stream import riscv_rand_instr_stream
+from pygen_src.isa.riscv_instr import riscv_instr_ins
+from pygen_src.riscv_instr_pkg import riscv_reg_t
+from pygen_src.target.rv32i import riscv_core_setting as rcs
+from pygen_src.riscv_pseudo_instr import riscv_pseudo_instr
+
+
+class riscv_directed_instr_stream(riscv_rand_instr_stream):
+
+    def __init__(self):
+        super().__init__()
+        self.name = ""
+        self.label = ""
+
+    def post_randomize(self):
+        for i in range(len(self.instr_list)):
+            self.instr_list[i].has_label = 0
+            self.instr_list[i].atomic = 1
+        self.instr_list[0].comment = "Start %0s" % (self.name)
+        self.instr_list[-1].comment = "End %0s" % (self.name)
+        if self.label != "":
+            self.instr_list[0].label = self.label
+            self.instr_list[0].has_label = 1
+
+
+class int_numeric_e(Enum):
```

Comment:
To be consistent throughout the whole implementation, would you mind changing the `Enum`s to `IntEnum` starting from 0? We need this for the functional coverage side as `Enums` is not supported (although this will probably get fixed later in pyvsc , I think there is no advantage of `Enum` over `IntEnum`. Please let me know if there are reasons on your end that make `Enum` a better option).

Reply:
Hi @hodjat91 
That's, make sense. I don't see any issue with using IntEnum.
I have changed it to IntEnum.